### ISSUE TITLE
Missing getByPressId() after categories moved to pkp-lib

### ISSUE THREAD

**Issue by tgc** [pkp/pkp-lib#4926]
Activating the 'Resize all existing cover thumbnails' checkbox in OMP 3.1.2-1 results in a stack trace from php:
```
[12-Jul-2019 13:23:30 Europe/Copenhagen] PHP Fatal error: Uncaught Error: Call to undefined method CategoryDAO::getByPressId() in /srv/omp/omp3-tgc/webroot/controllers/tab/settings/appearance/form/AppearanceForm.inc.php:105
Stack trace:
#0 /srv/omp/omp3-tgc/webroot/controllers/tab/settings/appearance/form/AppearanceForm.inc.php(79): AppearanceForm->_resizeCoverThumbnails(Object(Press), Object(CategoryDAO), '400', '566', '/srv/omp/omp3-t...')
#1 /srv/omp/omp3-tgc/webroot/lib/pkp/classes/controllers/tab/settings/SettingsTabHandler.inc.php(129): AppearanceForm->execute()
#2 /srv/omp/omp3-tgc/webroot/lib/pkp/classes/core/PKPRouter.inc.php(390): SettingsTabHandler->saveFormData(Array, Object(Request))
#3 /srv/omp/omp3-tgc/webroot/lib/pkp/classes/core/PKPComponentRouter.inc.php(257): PKPRouter->_authorizeInitializeAndCallRequest(Array, Object(Request), Array)
#4 /srv/omp/omp3-tgc/webroot/lib/pkp/classes/core/Dispatcher.inc.php(134): PKPComponentRouter->route(Object(Request))
#5 /srv/omp/omp3-tgc/webroot/lib/pkp/classes/core/PKPApplication.inc.php(252): Dispatcher->dispatc in /srv/omp/omp3-tgc/webroot/controllers/tab/settings/appearance/form/AppearanceForm.inc.php on line 105
```

When categories moved from omp to pkp-lib in pkp/pkp-lib#4158 the getByPressId function was left out presumably because it is specific to OMP.
However the OMP specific DAOs still have this function and OMP expects to be able to call it also on the CategoryDAO.

**Comment by asmecher** [comment]
@tgc, your suspicion is correct, we'd rather not add OMP-specific language to the pkp-lib repository. This PR should do it...

PR: https://github.com/pkp/omp/pull/692

Can you try it and confirm?

**Comment by tgc** [comment]
It does not work as is since AppearanceForm.inc.php still calls getByPressId when doing thumbnail resize.
I changed it to use getByContextId since I assume that was the point of adding it and then it works.

**Comment by asmecher** [comment]
Of course, @tgc, I got too focused on the implications and missed the goal. I've merged that for `stable-3_1_2` for the `3.1.2-2` release, and will forward-port to master. Thanks for your assistance!
